src/main/java/mrjake/aunis/tileentity/stargate/StargateAbstractBaseTile.java

```diff
@@ -1043,6 +1038,15 @@ public final void updateMergeState(boolean shouldBeMerged, EnumFacing facing) {
 		else {
 			onGateMerged();
 		}
+
+		// If the gate has already been merged, there is no need to merge it again.
+		// However, when the gate position changes, the members base pos should be updated as well.
```

Comment:
`when the gate position changes`
But it will be updated every time when `updateMergeState` will be called with `(shouldBeMerged && this.isMerged) == true`. Should be fine anyway? :I
Also you can move this stuff higher probably (above `if (!shouldBeMerged)` condition, line 1029)

Reply:
> But it will be updated every time when updateMergeState will be called with (shouldBeMerged && this.isMerged) == true. Should be fine anyway? :I

If that's the case, this means that the gate changed its position. There is no other code path that could trigger the if statement.

> Also you can move this stuff higher probably (above if (!shouldBeMerged) condition, line 1029)
I remember that I tested this, but something was wrong, I don't remember exactly what.